src/raster/r.connectivity/r.connectivity.network/r.connectivity.network.py

```diff
@@ -464,1220 +413,1141 @@ def main():
-        utils.install_packages(StrVector(missing_packages))
-    elif missing_packages:
-        grass.fatal(
-            "The following required R-packages {} are issing \
-                    on your system. Please install \
-                    them.".format(
-                ",".join(missing_packages)
-            )
+    # Check igraph version
+    if igraph.__version_info__ < (0, 6, 2):
+        gscript.fatal(
+            _(
+                "The required igraph version is 0.6.2 or \
+                     later, the installed version is {}. \
+                     Please download and install at least igraph \
+                     version 0.6.2"
+            ).format(".".join(igraph.__version_info__))
         )
 
-    # Check igraph version
-    rscript = """if(as.double(substr(
-                    packageDescription("igraph",
-                                       fields = c("Version")),1,3)
-                    )<=0.6) packageDescription("igraph",
-                                               fields = c("Version"))"""
-    igraph_version = robjects.r(rscript)
-
-    if not isinstance(igraph_version, rpy2.rinterface.RNULLType):
-        if i_flag:
-            grass.warning(
-                "The required igraph version is 0.6-2 or \
-                          later, the installed version is {}. \
-                          Installing latest version of igraph on \
-                          request.".format(
-                    igraph_version
-                )
-            )
-            utils.install_packages("igraph")
-        else:
-            grass.fatal(
-                "The required igraph version is 0.6-2 or \
-                         later, the installed version is {}. \
-                         Please download and install at least igraph \
-                         version 0.6-2".format(
-                    igraph_version
-                )
-            )
+    driver = db_connection["driver"]
+    database = db_name
+    schema = db_connection["schema"]
+    command = command
+
+    gscript.verbose(_("Reading and preparing input data...\n"))
+    vertices = np.gefromtext(
+        BytesIO(
+            gscript.read_command(
+                "v.db.select",
+                flags="c",
+                map=in_vertices,
+                columns="cat,{}".format(pop_proxy),
+                separator=",",
+            ).strip("\n")
+        ),
+        dtype=None,
+        names=["patch_id", "pop_proxy"],
+        delimiter=",",
+    )
 
-    rscript = """
-########################################################################
-###Preparation
-########################################################################
-verbose <- {}
-
-if (verbose) {{
-cat("Loading variables and packages...\n")
-}}
-
-library(igraph)
-library(nlme)
-library(codetools)
-library(rgrass7)
-library(DBI)
-""".format(
-        "{}".format(verbose).upper()
+    # Read edge data
+    edges = np.gefromtext(
+        BytesIO(
```

Comment:
```suggestion
        StringIO(
```
Following example in https://numpy.org/doc/stable/user/basics.io.genfromtxt.html